src/fromager/sources.py

```diff
@@ -151,6 +166,24 @@ def download_url(destination_dir: pathlib.Path, url: str) -> pathlib.Path:
     return outfile
 
 
+def _apply_template(templates: list[str] | None, req: Requirement, version: str):
+    if not templates:
+        return None
+
+    template_env = {"version": version}
+    if req.url and req.url.startswith(f"git+{GITHUB_URL}"):
+        tag = req.url.split("@")[-1]
```

Comment:
I think the above few lines are extracting a tag from a git URL, is that right? What if the URL only points to a branch and doesn't have a tag?

Reply:
Looking at the downstream plugins, i didn't find such a case. We can add it later if needed?

Reply:
I was thinking you probably want a little error handling here for that case. Even if it just raises an exception with a more meaningful error, that would be good.

Reply:
@tiran I was making the assumption that when a user defines a URL that uses tags for some package, then they know that the package will contain a URL which has tags (i.e. they have some background info on the package). Verifying whether it is a branch or tag might be quite a heavy operation.

Reply:
@tiran had a discussion with @dhellmann. We won't be implementing support for `tags` cause it will be removed from the downstream plugins. Plus like you said tags can't be reliable